core/src/main/clojure/xtdb/expression/temporal.clj

```diff
@@ -103,15 +104,15 @@
   (if (= ts-unit :second)
     `(.toEpochSecond ~form)
     `(let [form# ~form]
-       (Math/addExact (Math/multiplyExact (.toEpochSecond form#) ~(types/ts-units-per-second ts-unit))
-                      (quot (.getNano form#) ~(quot (types/ts-units-per-second :nano) (types/ts-units-per-second ts-unit)))))))
+       (Math/addExact (Math/multiplyExact (long (.toEpochSecond form#)) ~(types/ts-units-per-second ts-unit))
```

Comment:
Do test it, but as per the reflection thread, I think you can get away with `~(with-tag expr 'long)` for these (adds a type hint), rather than introducing an explicit cast. 

Arguably a nitpick, but just FYI.